apache/trafficcontrol
# Prohibit adding a cache group with no servers to a topology

## What does this PR (Pull Request) do?


- [x] This PR fixes #5143
- This PR adds validation to require that each cachegroup in a topology contains at least 1 server.

## Which Traffic Control components are affected by this PR?


- Traffic Ops

## What is the best way to verify this PR?

- Run the Traffic Ops API tests
- Try to create a topology that contains a cachegroup with no servers in it, make sure the POST fails
- Try to update a topology that contains a cachegroup with no servers in it, make sure the PUT fails

## The following criteria are ALL met by this PR
